bin/clinvar_jsons/traits_to_zooma_format.py

```diff
@@ -0,0 +1,207 @@
+import argparse
+import gzip
+import itertools
+import json
+from functools import lru_cache
+import sys
+from time import gmtime, strftime
+
+import progressbar
+import requests
+
+from clinvar_jsons_shared_lib import clinvar_jsons, get_traits_from_json, has_allowed_clinical_significance
+
+
+DATE = strftime("%d/%m/%y %H:%M", gmtime())
+
+
+def main(args):
+    with open(args.outfile_path, "wt") as outfile:
+        outfile.write("STUDY\tBIOENTITY\tPROPERTY_TYPE\tPROPERTY_VALUE\tSEMANTIC_TAG\tANNOTATOR\tANNOTATION_DATE\n")
+        line_count = file_len(args.infile_path)
+        bar = progressbar.ProgressBar(max_value=line_count,
+                                      widgets=[progressbar.AdaptiveETA(samples=1000)])
+        is_zooma_mapping_dict = {}
+        for clinvar_json in bar(clinvar_jsons(args.infile_path)):
+            if not has_allowed_clinical_significance(clinvar_json):
+                continue
+            process_clinvar_json(clinvar_json, outfile, args.zooma_host, args.filters,
+                                 is_zooma_mapping_dict)
```

Comment:
It needs to be provided as an argument because it is required to persist between function calls. A boolean value is stored for each trait name, specifying whether a trait name has any Zooma mappings (https://github.com/tomdcsmith/eva-cttv-pipeline/blob/f8b2c7444044e88043033b3c592415daf1794eaa/bin/clinvar_jsons/traits_to_zooma_format.py#L51). If a trait name has already been checked then that trait name doesn't need to be processed the next time it's seen (https://github.com/tomdcsmith/eva-cttv-pipeline/blob/f8b2c7444044e88043033b3c592415daf1794eaa/bin/clinvar_jsons/traits_to_zooma_format.py#L45).